react/RainbowLogo/RainbowLogo.js

```diff
@@ -0,0 +1,84 @@
+import styles from './RainbowLogo.less';
+import RainbowLogoText from './RainbowLogoText';
+import RainbowLogoIcon from './RainbowLogoIcon';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import classnames from 'classnames';
+
+const viewportDimensions = type => ({
+  default: { height: 68.031, width: 170.079 },
+  compact: { height: 68.031, width: 68.031 }
+})[type];
+
+const getViewPortDimension = (type, dimension) => viewportDimensions(type)[dimension];
+
+export default function Logo({
+  svgClassName,
+  invert,
+  compact,
+  textClass,
+  iconClass,
+  ...restProps }) {
+  const svgClasses = classnames(svgClassName, {
+    [styles.root]: true,
+    [styles.invert]: invert
+  });
+  const type = compact && 'compact' || 'default';
+  const height = getViewPortDimension(type, 'height');
+  const width = getViewPortDimension(type, 'width');
+
+  return (
+    <div {...restProps}>
+
```

Comment:
Nitpick, but this is an extra line-break compared to the original `Logo` component, which makes comparison a little harder.